tracpro/contacts/models.py

```diff
@@ -97,10 +100,12 @@ def kwargs_from_temba(cls, org, temba_contact):
 
         org_region_uuids = [r.uuid for r in Region.get_all(org)]
         region_uuids = intersection(org_region_uuids, temba_contact.groups)
-        region = Region.objects.get(org=org, uuid=region_uuids[0]) if region_uuids else None
-
-        if not region:  # pragma: no cover
-            raise ValueError("No region with UUID in %s" % ", ".join(temba_contact.groups))
+        if len(region_uuids) > 0:
+            region = Region.objects.filter(org=org, uuid__in=region_uuids).first() if region_uuids else None
+            if not region:  # pragma: no cover
```

Comment:
Ok, So there is also the other assumption made that region_uuids list will
always have one item or that the first item in that list is the only
relevant one from the query `region = Region.objects.get(org=org,
uuid=region_uuids[0])`  I also didn't understand the reason behind that and
that is why I changed it to `region = Region.objects.filter(org=org,
uuid__in=region_uuids).first()`. But I agree creating a new exception was
not very useful, just adding the contact id to the previous exception would
have done the trick.

On Sat, Apr 11, 2015 at 1:34 PM, Rowan Seymour <notifications@github.com>
wrote:

> In tracpro/contacts/models.py
> <https://github.com/ewheeler/tracpro/pull/1#discussion_r28194031>:
>
> > @@ -97,10 +100,12 @@ def kwargs_from_temba(cls, org, temba_contact):
> >
> >          org_region_uuids = [r.uuid for r in Region.get_all(org)]
> >          region_uuids = intersection(org_region_uuids, temba_contact.groups)
> > -        region = Region.objects.get(org=org, uuid=region_uuids[0]) if region_uuids else None
> > -
> > -        if not region:  # pragma: no cover
> > -            raise ValueError("No region with UUID in %s" % ", ".join(temba_contact.groups))
> > +        if len(region_uuids) > 0:
> > +            region = Region.objects.filter(org=org, uuid__in=region_uuids).first() if region_uuids else None
> > +            if not region:  # pragma: no cover
>
> No I think a contact should have a region, but if you want to know which
> contact is failing - just add the contact id to the original error message.
> I'm pretty sure that with this change, the exception on line 106 will never
> be raised.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/ewheeler/tracpro/pull/1/files#r28194031>.
>